### DISCUSSION TITLE
Getting "msg"="failed to record" "error"="observable instrument not registered for callback"  error

### DISCUSSION THREAD

**Discussion by karthi8185** [open-telemetry/opentelemetry-go#4239] [General]
Hi,
 I am getting ""msg"="failed to record" "error"="observable instrument not registered for callback"  while using the below code. 

Kindly let me know, for what kind of situation, this error may arise. 

Same Code:

for _, rowMetric := range data.Value {                                  
                                                                  
                rowName := rowMetric.RowName                                    
                for _, value := range rowMetric.Values {                        

                        fullName = fmt.Sprintf(tableName + "/" + rowName + "/" + metricName)
                        name, _ = meter.Int64ObservableGauge(                   
                                fullName,                                       
                        )                                                       
                        fullName1 = append(fullName1, fullName)                 
                        name1 = append(name1, name)                             
                        setMetricValue(fullName, int64(value.Value))            
                        count += 1                                              
                }                                  

              var out []metric.Observable                                     
                for _, inst := range name1 {                                    
                        if in, ok := inst.(wrappedc); ok {                      
                                out = append(out, in.unwrap())                  
                        } else {                                                
                                out = append(out, inst)                         
                        }                                                       
                }                                                               
                go func() {                                                     
                _, err = meter.RegisterCallback(                                
                        func(ctx context.Context, o metric.Observer) error {    
                                for travel := 0; travel < count; travel++ {     
                                        o.ObserveInt64(name1[travel], int64(getMetricValue(fullName1[travel])))
                                }                                               
                                return nil                                      
                        },                                                      
                        out...,                                                 
                )                                                               
                if err != nil {                                                 
                        handleErr(err, fmt.Sprintf("Unable to Register  the callback: %s", tableName))
                }                                                               
                                                                  
                }()

**Comment by pellared** [comment]
1. The snippet is not a working example
2. Why are you calling `RegisterCallback` in a separate goroutine?
